packages/office-ui-fabric-react/src/components/FocusZone/FocusZone.tsx

```diff
@@ -232,6 +250,112 @@ export class FocusZone extends BaseComponent<IFocusZoneProps, {}> implements IFo
+   *
+   * Note: Attempting to move this to a React event, rather than using a
+   * native event, broke the implementation.
+   */
+  private _onBlurCapture(ev: FocusEvent) {
+    const { current: root } = this._root;
+    const { target: from } = ev;
+    const isFocusDisappearing = ev.relatedTarget === null || ev.relatedTarget === document;
+
+    // If we are tabbing from parked state, reset parked.
+    if (from === root) {
+      this._setParkedFocus(false);
+    }
+
+    if (isFocusDisappearing) {
+      // The element may be removed; we need remember its index path and asynchronously
+      // re-evaluate if the item was removed. Blurs can also happen if the user just
+      // clicks the location bar, so we need to disambiguate.
+      const indexPath = this._getIndexPath(from as HTMLElement);
+
+      this._async.setTimeout(() => {
+        if (!elementContains(root, from as HTMLElement)) {
+          const nextFocusedElement = this._getElementByIndexPath(indexPath);
+
+          if (nextFocusedElement) {
+            // Try to move focus to the next one.
+            this._setActiveElement(nextFocusedElement, true);
+            nextFocusedElement.focus();
+          } else {
+            // There are no more elements. Park focus on the container.
+            this._setParkedFocus(true);
+          }
+        }
+      }, 0);
+    }
+  }
+
+  private _setParkedFocus(isParked: boolean): void {
+    const { current: root } = this._root;
+    if (root) {
+      if (isParked) {
+        if (!this.props.allowFocusRoot) {
+          this._parkedTabIndex = root.getAttribute('tabindex');
+          root.setAttribute('tabindex', '-1');
+        }
+        root.focus();
+      } else {
+        if (!this.props.allowFocusRoot) {
+          if (this._parkedTabIndex !== null) {
+            root.setAttribute('tabindex', this._parkedTabIndex);
+          } else {
+            root.removeAttribute('tabindex');
+          }
+        }
+      }
+    }
+  }
+
+  private _getElementByIndexPath(path: number[]): HTMLElement | undefined {
+    let element = this._root.current as HTMLElement;
+
+    for (const index of path) {
+      const nextChild = element.children[Math.min(index, element.children.length - 1)] as HTMLElement;
+
+      if (!nextChild) {
+        break;
+      }
+      element = nextChild;
+    }
+
+    element = isElementTabbable(element)
+      ? element
+      : getNextElement(this._root.current as HTMLElement, element, true) || getPreviousElement(this._root.current as HTMLElement, element)!;
+
+    return element as HTMLElement;
+  }
+
+  private _getIndexPath(toElement: HTMLElement): number[] {
+    const path: number[] = [];
+    const { current: root } = this._root;
+
+    while (toElement !== root) {
+      const parent = getParent(toElement);
+
+      if (parent === null) {
+        return [];
+      }
+
+      path.unshift(Array.prototype.indexOf.call(parent.children, toElement));
```

Comment:
I think some comments here or added for the function would help, or the name is throwing me off. It appears this function is traversing from `toElement` to root and then pushing them in the front of the array, but I'm not sure "what" it's doing or if that's correct...

Reply:
it's correct, i promise :) will add some comments. maybe this actually belongs in dom.ts.

Reply:
moved to utils, comments added, tests added.